bin/rector

```diff
@@ -209,3 +218,33 @@ final class RectorConfigsResolver
         return $this->configResolver->getFirstResolvedConfig();
     }
 }
+
+class PhpVersionChecker
+{
+    public static function check(): void
+    {
+        $minimumPhpVersion = self::getPhpVersionFromComposerJson();
+
+        $installedPhpVersion = PHP_MAJOR_VERSION . '.' . PHP_MINOR_VERSION . '.' . PHP_RELEASE_VERSION;
+
+        // Check minimum required PHP version
+        if (version_compare(PHP_VERSION, $minimumPhpVersion, '<')) {
+            throw new PHPVersionException(sprintf(
+                'PHP version %s or higher is required, but you currently have %s installed.',
+                $minimumPhpVersion,
+                $installedPhpVersion
+            ));
+        }
+    }
+
+    /**
+     * @return string
+     * @throws \Nette\Utils\JsonException
+     */
+    private static function getPhpVersionFromComposerJson(): string
+    {
+        $composerJson = FileSystem::read(__DIR__ . '/../composer.json');
+        $composerArray = Json::decode($composerJson, Json::FORCE_ARRAY);
+        return Strings::trim($composerArray['require']['php'], '~^>=*.');
```

Comment:
This won't work versions with `*` in the beginning or in the middle (e.g. 7.*.0, *.3.5) and with `<=`/`<` in the beginning but I believe it shouldn't be a problem.

Reply:
Could you create a basic test for this? Just so we have it covered.

Reply:
Can you please advise where is the best place for this kind of test?

Reply:
From top of my head: 
- src - `Rector\Core\Configuration\MinimalVersionChecker`
- tests - `Rector\Core\Tests\Configuration\Minimal\VersionCheckerTest`